src/Entity/Adherent.php

```diff
@@ -789,6 +828,19 @@ public function setCoordinatorManagedAreaCodesAsString(string $codes = null): vo
         return $this->memberships;
     }
 
+    final public function getGroupMemberships(): GroupMembershipCollection
+    {
+        if ($this->groupMemberships instanceof Collection) {
+            if (!$this->groupMemberships instanceof GroupMembershipCollection) {
+                $this->groupMemberships = new GroupMembershipCollection($this->groupMemberships->toArray());
+            }
+        } else {
+            $this->groupMemberships = new GroupMembershipCollection((array) $this->groupMemberships);
```

Comment:
What's supposed to be casted here? I don't think this is necessary to handle `iterable` like this. An `Iterator` would have to pas passed to `iterator_to_array`.
I suggest to remove this overingeenering.